adapters/repos/db/index.go

```diff
@@ -2323,13 +2325,14 @@ func (i *Index) getOptInitLocalShard(ctx context.Context, shardName string, ensu
 		return nil, func() {}, errAlreadyShutdown
 	}
 
-	// make sure same shard is not inited in parallel
-	i.shardCreateLocks.Lock(shardName)
-	defer i.shardCreateLocks.Unlock(shardName)
+	// make sure same shard is not inited in parallel. In case it is not loaded yet, switch to a RW lock and initialize
+	// the shard
+	i.shardCreateLocks.RLock(shardName)
 
 	// check if created in the meantime by concurrent call
 	shard = i.shards.Load(shardName)
```

Comment:
this lock: `i.shardCreateLocks.RLock(shardName)` is only needed to acquire shard here?

bc if yes, then we could do this:
```go
func() {
    i.shardCreateLocks.RLock(shardName)
    defer i.shardCreateLocks.RUnlock(shardName)
    shard = i.shards.Load(shardName)
}()
```
and remove the [else clause](https://github.com/weaviate/weaviate/pull/9867/files#diff-d080e5efa621477a1d974bced14c39a8912fcb7a649eb4a0fb219647728dbc3dR2360)